mvrx/src/main/kotlin/com/airbnb/mvrx/MvRxViewModelStore.kt

```diff
@@ -103,9 +103,12 @@ class MvRxViewModelStore(private val viewModelStore: ViewModelStore) {
             } else {
                 ownerArgs
             }
-            map[it] = (host as? Fragment)
-                    ?.let { fragment -> restoreViewModel(fragment, viewModelsState.getParcelable(it), arguments) }
-                    ?: restoreViewModel(host as FragmentActivity, viewModelsState.getParcelable(it), arguments)
+            val holder = viewModelsState.getParcelable<MvRxPersistedViewModelHolder>(it)!!
```

Comment:
Since a null value will crash anyway, can you give it a good error message with `?: throw IllegalStateException(...)`?

Reply:
Makes sense, will do.